app/src/test/java/com/codesoom/http/HttpRequestTest.java

```diff
@@ -0,0 +1,51 @@
+package com.codesoom.http;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class HttpRequestTest {
+    @DisplayName("getLongFromPathParameter(idx): HttpRequest의 path에서 지정한 위치(idx)의 값을 Long 타입으로 반환한다")
+    @Test
+    void getLongFromPathParameter_success() {
+        HttpRequest request = new HttpRequest(HttpMethod.GET, "/tasks/1", "");
+
+        Long id = request.getLongFromPathParameter(2);
+        assertThat(id).isEqualTo(1L);
+    }
+
+    @DisplayName("getLongFromPathParameter(idx): HttpRequest의 path에서 지정한 위치에(idx) 값이 없을 경우 예외가 발생한다.")
+    @Test
+    void getLongFromPathParameter_fail_noValue() {
+        HttpRequest request = new HttpRequest(HttpMethod.GET, "/tasks/", "");
+
+        assertThrows(IllegalArgumentException.class, () -> {
+                    request.getLongFromPathParameter(2);
+                }
+        );
+    }
+
+    @DisplayName("getLongFromPathParameter(idx): HttpRequest의 path에서 지정한 위치(idx)가 없는 경우 예외가 발생한다.")
+    @Test
+    void getLongFromPathParameter_fail_noPlace() {
+        HttpRequest request = new HttpRequest(HttpMethod.GET, "/tasks", "");
+
+        assertThrows(IllegalArgumentException.class, () -> {
+                    request.getLongFromPathParameter(2);
+                }
```

Comment:
`예외가 발생한다`는 표현과 `throws`, `assertThrows`에 대해 생각해 봅시다. 애플리케이션을 사용하는 사람 입장에서 예외는 "발생하는" 것일 거에요. 하지만 애플리케이션을 코딩해서 만드는 사람 입장에서 생각해보면 어떤 상황에서 예외를 `throws` 할지를 정해주고 있다는 것을 알 수 있습니다. 우리처럼 만드는 입장에서 예외는 발생해서 발견되는 것이어서는 곤란할 수 있어요. 예외는 던지는 거죠.

```suggestion
    @DisplayName("getLongFromPathParameter(idx): HttpRequest의 path에서 지정한 위치(idx)가 없는 경우 예외를 던진다.")
    @Test
    void getLongFromPathParameter_fail_noPlace() {
        HttpRequest request = new HttpRequest(HttpMethod.GET, "/tasks", "");

        assertThrows(IllegalArgumentException.class, () -> {
                    request.getLongFromPathParameter(2);
                }
```

Reply:
별로 깊게 생각해보지 못했는데 리뷰 감사합니다!!